src/progress-indicator/progress-indicator.tsx

```diff
@@ -151,34 +74,56 @@ export const ProgressIndicator = <T,>({
                     id={getId(stepIndex, currentIndex)}
                 >
                     <IndicatorBar highlighted={highlighted}></IndicatorBar>
-                    <IndicatorTitle
+                </Indicator>
+            );
+        });
+    };
+
+    const renderStepTitleDesktop = () => {
+        return steps.map((step: T, stepIndex: number) => {
+            const highlighted = stepIndex <= currentIndex;
+            const fontWeight = stepIndex === currentIndex ? "bold" : "regular";
+
+            return (
+                <Indicator
+                    key={stepIndex}
+                    aria-label={getAriaLabel(stepIndex, currentIndex)}
+                    id={getId(stepIndex, currentIndex)}
+                >
+                    <IndicatorTitleDesktop
                         highlighted={highlighted}
                         weight={fontWeight}
                     >
                         {getDisplayValue(step)}
-                    </IndicatorTitle>
+                    </IndicatorTitleDesktop>
                 </Indicator>
             );
         });
     };
 
-    const renderFade = () => {
+    const renderStepTitleMobile = () => {
         return (
-            <>
-                {showFadeLeft && (
-                    <Fade backgroundColor={fadeColor} position={"left"} />
-                )}
-                {showFadeRight && (
-                    <Fade backgroundColor={fadeColor} position={"right"} />
-                )}
-            </>
+            <Indicator
+                key={currentIndex}
+                aria-label={getAriaLabel(currentIndex, currentIndex)}
+                id={getId(currentIndex, currentIndex)}
```

Comment:
noticed that the id is set on the bar and on the title text. should try to keep it unique, I would suggest to append a `-title` for the title